sysl2/sysl/integration/deps_test.go

```diff
@@ -0,0 +1,399 @@
+		{
+			Stmt: &sysl.Statement_Foreach{
+				Foreach: &sysl.Foreach{
+					Stmt: []*sysl.Statement{},
+				},
+			},
+		},
+		{
+			Stmt: &sysl.Statement_Group{
+				Group: &sysl.Group{
+					Stmt: []*sysl.Statement{},
+				},
+			},
+		},
+		{
+			Stmt: &sysl.Statement_Alt{
+				Alt: &sysl.Alt{
+					Choice: []*sysl.Alt_Choice{
+						{
+							Stmt: []*sysl.Statement{},
+						},
+					},
+				},
+			},
+		},
+	}
+	cs := NewCallSlice()
+	expected := []*sysl.Call{
+		{
+			Target: &sysl.AppName{
+				Part: []string{"AppA"},
+			},
+			Endpoint: "EndptA",
+		},
+		{
+			Target: &sysl.AppName{
+				Part: []string{"AppB"},
+			},
+			Endpoint: "EndptB",
+		},
+	}
+
+	// When
+	cs.CollectCallStatements(stmts)
+
+	// Then
+	assert.Equal(t, cs.GetSlice(), expected)
+}
+
+var mod = &sysl.Module{
+	Apps: map[string]*sysl.Application{
+		"AppA": {
+			Attrs: map[string]*sysl.Attribute{},
+		},
+		"AppB": {
+			Attrs: map[string]*sysl.Attribute{},
+			Endpoints: map[string]*sysl.Endpoint{
+				"EndptB": {
+					Stmt: []*sysl.Statement{
+						{
+							Stmt: &sysl.Statement_Call{
+								Call: &sysl.Call{
+									Target: &sysl.AppName{
+										Part: []string{"AppE"},
+									},
+									Endpoint: "EndptE",
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Action{
+								Action: &sysl.Action{
+									Action: "Get",
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Ret{
+								Ret: &sysl.Return{
+									Payload: "Return A",
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Cond{
+								Cond: &sysl.Cond{
+									Stmt: []*sysl.Statement{},
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Loop{
+								Loop: &sysl.Loop{
+									Stmt: []*sysl.Statement{},
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_LoopN{
+								LoopN: &sysl.LoopN{
+									Stmt: []*sysl.Statement{},
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Foreach{
+								Foreach: &sysl.Foreach{
+									Stmt: []*sysl.Statement{},
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Group{
+								Group: &sysl.Group{
+									Stmt: []*sysl.Statement{},
+								},
+							},
+						},
+						{
+							Stmt: &sysl.Statement_Alt{
+								Alt: &sysl.Alt{
+									Choice: []*sysl.Alt_Choice{
+										{
+											Stmt: []*sysl.Statement{},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"AppC": {
+			Attrs: map[string]*sysl.Attribute{},
+		},
+		"AppD": {
+			Attrs: map[string]*sysl.Attribute{},
+		},
+		"AppE": {
+			Attrs: map[string]*sysl.Attribute{},
+			Endpoints: map[string]*sysl.Endpoint{
+				"EndptE": {
+					Stmt: []*sysl.Statement{},
+				},
+			},
+		},
+	},
+}
+
+func TestFindHighlightApps(t *testing.T) {
+	// Given
+	ds := NewDependencySet()
+	s := MakeAppElement("AppA", "EndptA")
+	tar := MakeAppElement("AppB", "EndptB")
+	dep := MakeAppDependency(s, tar)
```

Comment:
rewrite this as
```go
	dep := MakeAppDependency(MakeAppElement("AppA", "EndptA"), MakeAppElement("AppB", "EndptB"))
```
as you are not referring to these variables (s, tar).

Reply:
got it. fixed.